### DISCUSSION TITLE
Production usage of ART?

### DISCUSSION THREAD

**Discussion by dping1** [Trusted-AI/adversarial-robustness-toolbox#1455] [General]
Hi, does anyone know if ART has been used by any organizations in a production environment?

**Comment by beat-buesser** [comment]
Hi @dping1 That's a great question! I have heard of start-ups building their products with ART and larger organisations exploring applications with ART. May I ask why you are asking, maybe I can help further?

**Reply by dping1** [reply]
HI @beat-buesser, thank you for your response. I was asking because I was interested to know if the ART library can scale for large datasets. When I tried some of the samples, it seemed to take a long time to generate adversarial samples on a single machine.  Can ART be distributed across multiple nodes and run in parallel?

**Reply by beat-buesser** [reply]
Hi @dping1 So far, ART does not itself provide any tools to setup distributed or parallel computing. But ART aims to, and in some ML frameworks already can, take full advantage of parallel computing capabilities provided by the ML libraries to evaluate the models, calculate their loss gradients, etc. on multiple GPUs, in these cases the parallel setup is entirely defined in the model setup outside of ART.
I'm hesitating to officially say that all parallel computing capabilities in all frameworks are already support because we have not tested all options yet, but we definitely aim to support them. In case you would discover any limitation in this area let us know.
For the acceleration of attack algorithms around the model evaluation the best strategy is to run multiple instances of the attack in parallel to generate a large number of adversarial examples.

Which attack are you running?
